services/src/main/java/org/keycloak/models/workflow/WorkflowsManager.java

```diff
@@ -343,6 +340,26 @@ public Workflow toModel(WorkflowRepresentation rep) {
         return workflow;
     }
 
+    private void validateWorkflow(WorkflowRepresentation rep, MultivaluedHashMap<String, String> config) {
+        // Validations:
+        //  immediate workflow cannot have time conditions
+        //  scheduled workflow must have time conditions
+        //  workflow cannot be both immediate and recurring
+        boolean isImmediate = config.containsKey("scheduled") && !Boolean.parseBoolean(config.getFirst("scheduled"));
+        boolean isRecurring = config.containsKey("recurring") && Boolean.parseBoolean(config.getFirst("recurring"));
+        boolean hasTimeCondition = rep.getSteps().stream().allMatch(step -> step.getConfig() != null
+                && step.getConfig().containsKey("after"));
```

Comment:
Better refence "after" using `org.keycloak.models.workflow.WorkflowStep#AFTER_KEY`. I would also introduce a constant for the others we are missing.